### PR TITLE
configure: sort print_pkgs output, shellcheck

### PR DESCRIPTION
**Pull request by classabbyamp**
This adds a dependency-order-sorted version of the `print_pkgs` target, which generates `pkgs-sorted.txt` containing the list of packages in the order that make would build them for the new buildbot setup.

This change should be backwards compatible.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -188,33 +187,57 @@ touch Makefile
...
+
+built/%: tobuild/%
+	@echo "[xbps-src]	\${@F}"
+	@( $XSC pkg \${@F}; rval=\$\$?; [ \$\$rval -eq 2 ] && exit 0 || exit \$\$rval )
+	@touch \$@
+	@rm tobuild/\${@F}
+endif
+
+# Dependencies
+EOM
```
Comment (classabbyamp) [inline configure:228]: the core functional changes are in this heredoc, with the `SORTONLY` variable and targets

**Comment by Duncaen** [comment]
kinda want to get rid of xbps-bulk, the build servers currently use the hacked in `xbps-src-make` which is faster and actually recursively sorts the whole chain. https://github.com/void-linux/xbps-bulk/commit/8fab58a62a8c26766f516de4d49f527cc6e1d200

**Comment by classabbyamp** [comment]
ah i did not know it was using xbps-src-make

**Comment by classabbyamp** [comment]
that's not checked into void-infra at all

**Pull request closed without merging**